ChainSafe/gossamer
# Implement prepared_block_for_proposing Telemetry Interface

This PR implements prepared_block_for_proposing Telemetry Interface.


## Changes



- Sends a telemetry message of type `prepared_block_for_proposing` which a block is built

## Tests


- Setup a local telemetry server using substrate-telemetry.

Use the branch `print-payload-type` from my fork `https://github.com/kishansagathiya/substrate-telemetry` (It print the message received, type of message and error. From what I have played around upstream doesn't do that.)

Run telementry backend (`telemetry_core` and `telemetry_shard`).

-  Run gossamer by setting telemetry  url to the local server. 

For this pull `kishansagathiya:issue-1502` (branch from https://github.com/ChainSafe/gossamer/pull/1890), since this feature is not merged at the time of writing.

Run `./bin/gossamer --chain dev --name 'Kishans Test Node' --telemetry-url 'ws://localhost:8001/submit 0'`

- Wait for message to get send and received. Follow logs of `telemetry_shard` command. After few minutes it should print output as follows
```
telemetry message received: {"hash":"0x5814aec3e28527f81f65841e034872f3a30337cf6c33b2d258bba6071e37e27c","msg":"prepared_block_for_proposing","number":"1","ts":"2021-10-18T20:41:49.787310312+05:30"}
message type Payload::PreparedBlockForProposing

```


```

```

## Issues



- Closes #1844

## Primary Reviewer 



- @noot
